### PR TITLE
An initial version of being able to select non-grain crop_type

### LINKED ISSUE #56: crop_type is too restrictive if we want to demonstrate with diverse weed datasets
restricted currently to grains... so we need at least a free text alternative for "other", e.g. an `other_crop_type` field

### PR DESCRIPTION
**Pull request by jnothman**
Fixes #56 by adding the option to enter the scientific name of a non-grain species, or "pasture".

Current solution is quite minimal:
* it still uses common, English-language names for grain crops, even in the metadata.
* it does little to validate the scientific name entered.
* visually it isn't great, and would be improved by #394 since currently I have no way to head these form elements with "Crop Type" which is really needed there.

image

image

### CODE REVIEW

File: search/src/Schemas/AgContext.json [hunk 1]
```diff
@@ -27,31 +27,46 @@
...
+            "lentils",
+            "field_peas",
+            "faba_beans",
+            "mung_beans",
+            "canola",
+            "mustard",
+            "flax",
+            "linseed",
+            "safflower",
+            "cotton"
```
Comment (geezacoleman) [inline search/src/Schemas/AgContext.json:53]: Cotton isn't technically a grain crop, more of a fibre crop but it's been there from the beginning so I think just leave it there for now.

**Review by geezacoleman — COMMENTED** [review]
I think this is a good interim option, but we'll need to be careful in the review phase to make sure we catch any scientific names for 'other crop' that don't follow convention.

**Comment by jnothman** [comment]
> I think this is a good interim option, but we'll need to be careful in the review phase to make sure we catch any scientific names for 'other crop' that don't follow convention.

I'm just trying to break down the task: next step is to have everything use a scientific name, under the hood at least.

So upcoming steps include:
* validating scientific names against EPPO or GBIF, in both crop type and 
* changing grain crop types to scientific name
* displaying hierarchical categories

I've tried playing with a hierarchical selection tool for species, but it's not really working to my satisfaction (when you search for an element by typing, you lose the hierarchy): https://2o1pv6925p.csb.app/

**Comment by jnothman** [comment]
So this PR is mostly about improving the public form and capability for other crop types, not about making the backend consistent.

**Pull request merged**